docs/ossm/ossm2-migration/multi-tenancy/README.md

```diff
@@ -0,0 +1,293 @@
+     values:
+       meshConfig:
+         discoverySelectors:
+           - matchLabels:
+               tenant: tenant-a
+     version: v1.23.0
+   ```
+
+1. Add your `tenant` label to each one of your dataplane namespaces.
+
+   With 2.x, we enrolled namespaces into the mesh by adding them to the Service Mesh Member Roll resource. In 3.x, you must label each one of your dataplane namespaces with this label. For every namespace in your Service Mesh Member Roll, add your tenant label to the namespace.
+
+   ```sh
+   oc label ns bookinfo tenant=tenant-a
+   ```
+
+   Now we are ready to migrate our workloads from our 2.x controlplane to our 3.x controlplane.
+
+#### Migrate Workloads
+
+1. Update injection labels on the dataplane namespace.
+
+   Here we're adding two labels to the namespace:
+
+   1. The `istio.io/rev: istio-tenant-a` label which ensures that any new pods that get created in that namespace will connect to the 3.0 proxy.
+   2. The `maistra.io/ignore-namespace: "true"` label which will disable sidecar injection for 2.6 proxies in the namespace. This ensures that 2.6 will stop injecting proxies in this namespace and any new proxies will be injected by the 3.0 controlplane. Without this, the 2.6 injection webhook will try to inject the pod and it will connect to the 2.6 proxy as well as refuse to start since it will have the 2.6 cni annotation.
+
+   **Note:** that once you apply the `maistra.io/ignore-namespace` label, any new pod that gets created in the namespace will be connected to the 3.0 proxy. Workloads will still be able to communicate with each other though regardless of which controlplane they are connected to.
+
+   ```sh
+   oc label ns bookinfo istio.io/rev=istio-tenant-a maistra.io/ignore-namespace="true" --overwrite=true
+   ```
+
+1. `curl` the productpage pod in `bookinfo` to ensure proxies can still communicate with one another.
+
+   ```sh
+   oc exec -it -n bookinfo deployments/productpage-v1 -c istio-proxy -- curl localhost:9080/productpage
+   ```
+
+   You should see
+
+   ```html
+   ...
+           <p>Absolutely fun and entertaining. The play lacks thematic depth when compared to other plays by Shakespeare.</p>
+           <small>Reviewer2</small>
+
+
+           <font color="black">
+             <!-- full stars: -->
+
+             <span class="glyphicon glyphicon-star"></span>
+
+             <span class="glyphicon glyphicon-star"></span>
+
+             <span class="glyphicon glyphicon-star"></span>
+
+             <span class="glyphicon glyphicon-star"></span>
+
+             <!-- empty stars: -->
+
+             <span class="glyphicon glyphicon-star-empty"></span>
+
+           </font>
+
+
+         </blockquote>
+
+         <dl>
+           <dt>Reviews served by:</dt>
+           <u>reviews-v2-6dd458b5db-frrlb</u>
+
+         </dl>
+   ...
+   ```
+
+1. Migrate workloads.
+
+   You can now restart the workloads so that the new pod will be injected with the 3.0 proxy.
```

Comment:
Do you think it makes sense to add a note about possible downtime for the apps? Something like: with correct replica configuration in your deployments, this operation has no traffic disruptions

Reply:
Is this the first time as an OSSM user you need to restart your workloads as part of the controlplane upgrade process? Restarting workloads is a normal kube process so I don't think we need to point this out unless we are going to give users advice on what a correct replica configuration is.